src/utils/permissions.ts

```diff
@@ -79,6 +80,20 @@ export const isAppExecutor = async (d2: D2) => {
     return globalAdmin || !!userRoles.find((role: any) => role.name === name);
 };
 
+export const verifyUserHasAccessToSyncRule = async (d2: D2, syncRuleUId: string) => {
```

Comment:
I believe we are not considering create action here.

SyncRuleId would be undefined when ``/sync-rules/:type/new`` and we would return true if user is ``appConfigurator``.

Reply:
Done! New routes was working but I have modified the code to more readable